### PR TITLE
Fix extensions not loading with code and file

### PR DESCRIPTION
**Pull request by kokofixcomputers**
With the new dashboard, extensions currently do not load with code due to an encoding issue. This pr solves that.

### CODE REVIEW

File: src/main/dashboard/app.tsx [hunk 1]
```diff
@@ -127,57 +134,73 @@ function LoaderForm() {
...
+                  if (fileContent()) { // Use stored file content
+                    await eureka.load(stringToDataURL(fileContent()));
+                  }
+                  break;
               }
-              break;
-          }
-        }}>
+            } catch (error) {
+              console.error("Error loading extension:", error); // Handle any errors that occur during loading
```
Comment (SimonShiki) [inline src/main/dashboard/app.tsx:201]: May we should show error below the "Load" button?

**Review by SimonShiki — CHANGES_REQUESTED** [review]

**Comment by kokofixcomputers** [comment]
Sure… I’ll do that now

--

File: src/main/patches/applier.ts [hunk 2]
```diff
@@ -46,12 +46,27 @@ const checkEureka = (eurekaFlag: string): boolean | null => {
  * @param {!string} opcode The opcode to examine for extension.
  * @return {?string} The extension ID, if it exists and is not a core extension.
  */
-function getExtensionIdForOpcode (opcode: string) {
+function getExtensionIdForOpcode(opcode) {
+    // Check if opcode is undefined or not a string
+    if (typeof opcode !== 'string' || opcode === undefined) {
```
Comment (SimonShiki) [inline src/main/patches/applier.ts]: `opcode === undefined` is unnecessary

File: src/main/patches/applier.ts [hunk 3]
```diff
@@ -46,12 +46,27 @@ const checkEureka = (eurekaFlag: string): boolean | null => {
...
+        return; // Return undefined if opcode is not valid
```
Comment (SimonShiki) [inline src/main/patches/applier.ts]: throw or return a empty string

File: src/main/dashboard/app.tsx (hunk 1 above)
Comment (kokofixcomputers) [reply src/main/dashboard/app.tsx:201]: I added it. I also fixed a bug where projects do not load when the project is saved without eureka

Comment (kokofixcomputers) [reply src/main/dashboard/app.tsx:201]: Should I go ahead and resolve this? Or do you have anything else to add?

File: src/main/dashboard/app.tsx [hunk 4]
```diff
@@ -127,59 +135,78 @@ function LoaderForm() {
...
+              setErrorMessage("Error loading extension:" + error);
+              console.error("Error loading extension:", error); // Handle any errors that occur during loading
+            } finally {
+              setLoading(false); // Ensure loading is set to false after loading completes or fails
+            }
+          }}
+        >
           <FormattedMessage id="eureka.loader.load" default="Load Extension" />
         </button>
+        {errorMessage() && <p style={{ color: 'red' }}>{errorMessage()}</p>} {/* Display error message below the button */}
```
Comment (SimonShiki) [inline src/main/dashboard/app.tsx]: use className + css + Scratch's red

**Review by SimonShiki — CHANGES_REQUESTED** [review]

File: src/main/dashboard/app.tsx (hunk 4 above)
Comment (kokofixcomputers) [reply src/main/dashboard/app.tsx:209]: I think I did. Also, please explain what you mean by `use className + css + Scratch's red`

File: src/main/patches/applier.ts (hunk 3 above)
Comment (kokofixcomputers) [reply src/main/patches/applier.ts:52]: Sure. But it seems to work now.

File: src/main/patches/applier.ts (hunk 2 above)
Comment (kokofixcomputers) [reply src/main/patches/applier.ts:51]: Okay

File: src/main/dashboard/app.tsx (hunk 4 above)
Comment (SimonShiki) [reply src/main/dashboard/app.tsx]: tsx
```typescript
<span className='errorText'>{errorMessage()}</span>
```

css file
```css
.errorText {
    font-size: 12px;
    color: var(--eureka-error-primary);
}
```

Comment (kokofixcomputers) [reply src/main/dashboard/app.tsx]: I added it. Please let me know if i added the css in the wrong file. style.module.css

Comment (SimonShiki) [reply src/main/dashboard/app.tsx]: Please check whether `var(--eureka-error-primary)` exists, and add that css variable in global.css if not

Comment (kokofixcomputers) [reply src/main/dashboard/app.tsx]: Done!

**Review by SimonShiki — APPROVED** [review]
LGTM

**Comment by kokofixcomputers** [comment]
Thanks!

**Pull request merged**